### PR TITLE
Rename specialist document

### PR DESCRIPTION
**Pull request by chrislo**
This PR replaces the use of the term "specialist document" with the term "section". The publishing of what are known as Specialist Documents was moved to a separate application [some time ago](). However the name still remains in this codebase and seems to be used to refer to a section of a manual. We've gone with `Section` as opposed to something like `ManualSection` as the application currently only has responsibility for manuals and their sections and this seems to match the language used in the UI. 

This PR does not currently touch any data - the underlying mongodb collection for `SectionEdition` is still `specialist_document_editions`. I'd like someone to help check that this continues to work with a recent copy of a production-like database. 

I think we'll rename the collection in a subsequent commit after this one has landed as we need to do some investigation about whether that would involve application downtime.

### CODE REVIEW

**Comment by chrisroos** [comment]
> This PR does not currently touch any data - the underlying mongodb collection for SectionEdition is still specialist_document_editions. I'd like someone to help check that this continues to work with a recent copy of a production-like database.

I took a recent dump of production data and checked out this branch locally. I clicked around the manuals-publisher UI and can confirm that everything appears to work as expected.

**Comment by chrislo** [comment]
Thanks @chrisroos - I've rebased this against master so I think it's ready for review.

**Comment by chrisroos** [comment]
I've added issue #848 to prompt us to rename the Mongo collection once this has been merged.

File: spec/repositories/versioned_manual_repository_spec.rb [hunk 1]
```diff
@@ -38,19 +38,19 @@
...
       end
 
-      it "has the first draft of the specialist document editions as Specialist Document instances attached" do
+      it "has the first draft of the section editions as Section instances attached" do
         documents = subject.documents.to_a
         expect(documents.size).to eq 2
 
         document_1 = documents[0]
-        expect(document_1).to be_a ::SpecialistDocument
+        expect(document_1).to be_a ::Section
```
Comment (floehopper) [inline spec/repositories/versioned_manual_repository_spec.rb]: Do you know why the `::` prefix is needed here and elsewhere in this file?

File: app/models/document_factory_registry.rb [hunk 2]
```diff
@@ -6,7 +6,7 @@
 require "builders/manual_document_builder"
 require "manual_with_documents"
 require "slug_generator"
-require "specialist_document"
+require "section"
```
Comment (floehopper) [inline app/models/document_factory_registry.rb]: Is this require statement still necessary now that the classname matches the filename?

File: app/models/attachment.rb [hunk 3]
```diff
@@ -7,7 +7,7 @@ class Attachment
   field :file_id, type: String
   field :file_url, type: String
 
-  embedded_in :specialist_document_edition
+  embedded_in :section_edition
```
Comment (floehopper) [inline app/models/attachment.rb]: I think it would be useful to mention why this change does not require any database migration, i.e. can you confirm that none of the existing attachments store a reference to their parent edition using this key. I'm guessing it's because this is an embedded document...?

File: app/repositories/repository_registry.rb [hunk 4]
```diff
@@ -1,13 +1,12 @@
 require "specialist_document_repository"
 require "manual_repository"
-require "specialist_document_edition"
+require "section_edition"
```
Comment (floehopper) [inline app/repositories/repository_registry.rb]: Is this require statement necessary now that the class name matches the filename?

**Comment by floehopper** [comment]
There are a bunch of classes named `*ManualSection*`. Given the choice to rename `SpecialistDocument` to `Section`, we will want to rename them to `*Section*`.

**Comment by chrisroos** [comment]
I'm going to rebase this on master given all the recent changes.

**Comment by chrisroos** [comment]
I've rebased and force pushed.

I'm going to hold off merging until I can pair with @chrislo to address @floehopper's comments and to double check that we're happy with the name `Section` instead of `ManualSection`.

File: app/models/attachment.rb (hunk 3 above)
Comment (chrislo) [inline app/models/attachment.rb]: Yep, that's correct. I've addressed this comment in an improved commit message on b0b409ad6.

**Comment by chrislo** [comment]
I've raised https://github.com/alphagov/manuals-publisher/issues/867 to make sure we keep track of the suggestion to rename `ManualSection` to `Section`

File: spec/repositories/versioned_manual_repository_spec.rb (hunk 1 above)
Comment (chrislo) [inline spec/repositories/versioned_manual_repository_spec.rb]: No, it struck us as odd too - will investigate after merging.

File: app/models/document_factory_registry.rb (hunk 2 above)
Comment (chrislo) [inline app/models/document_factory_registry.rb]: Possibly not - I think we'll take a look at this after we merge this PR, there's probably a few unnecessary requires scattered around.

**Comment by chrislo** [comment]
Rebasing against master and force pushing before merging.
